Readme.md

```diff
@@ -0,0 +1,112 @@
+    * [Cloud Build](https://cloud.google.com/build) build is created to generate and update docker images with application code we upload.
+    * These artifacts are then stored in [Artifact Registry](https://cloud.google.com/artifact-registry) after the build is run.
+    * Cloud Functions Gen 2 run on [Cloud Run](https://cloud.google.com/run) so a cloud run service is created for the application.
+* [Service Account](https://cloud.google.com/iam/docs/service-account-overview) for cloud functions (cloud run under the hood)
+* Permissions to interact with Google APIs to,
+    * Fetch secrets
+    * Write logs
+    * Fetch artifact from storage bucket
+    * Allow unauthenticated invocations; for Slack to send invoke cloud function with events (permission assigned to cloud run service)
+
+## CI/CD with Github Actions and Google Cloud Workload Identity Federation
+
+<!-- TODO -->
+<!-- Xplorersbot uses Github Actions for its CI/CD pipeline. With every push from every branch, unit tests will run and the workflow is defined in `.github/workflows/run-tests.yml`. -->
+
+Historically, applications running outside the Google Cloud had to rely on service account keys to access resources within the Google Cloud. However, as service account keys are long-lived credentials with permissions to interact and change the state of resources in the cloud, they pose a security threat if not managed appropriately.
+
+Fortunately, through [workload identity federation](https://cloud.google.com/iam/docs/workload-identity-federation), external identities can be granted IAM roles, including the capability to impersonate service accounts and obtain a short-lived token. This approach effectively eliminates the security and maintenance overhead associated with service account keys.
+
+[Service account impersonation](https://cloud.google.com/iam/docs/workload-identity-federation#impersonation) is used toobtain a short-lived OAuth 2.0 access token which lets github actions impersonate the service account to interact with Google Cloud APIs and deploy the application. Check out this [repository](https://github.com/xplorer-io/xplorers-gcloud-infra-bootstrap) for its implementation. Also see this [blog post](https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions) to understand the github actions workflow with Google Cloud.
+
+Together, [Github Actions](https://github.com/features/actions) and Terraform are being used to deploy XplorersBot to Google Cloud once the changes have been tested and peer reviewed and merged to the mainline branch.
+
+With every pull request merge to `main` branch, actions will automatically update the production environment by deploying to Google Cloud (workflow `.github/workflows/deploy-xplorers-bot.yml`)
+
+## Release tags
+
+With the help of github actions and release drafter tool, a new release tag is created on every merge to `main`.
```

Comment:
The readme is good and details, still some things I would suggest to be done:

1. Keep it simple, like answering 2 questions - what does the project do, how do I run it locally? Have the features included in the main readme file.
2. Break away the infra parts to a different file like `/docs/infra.md`
3. Other things can go into different md files too.

Reply:
I have broken down Readme and abstracted away a lot of the information in their own files